nvtabular/tools/dataset_inspector.py

```diff
@@ -0,0 +1,191 @@
+#
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import json
+
+import cudf
+import fsspec
+import numpy as np
+from dask.distributed import Client
+from dask_cuda import LocalCUDACluster
+
+from nvtabular.io import Dataset
+
+
+# Class to help Json to serialize the data
+class NpEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, np.integer):
+            return int(obj)
+        elif isinstance(obj, np.floating):
+            return float(obj)
+        elif isinstance(obj, np.bool_):
+            return bool(obj)
+        elif isinstance(obj, np.ndarray):
+            return obj.tolist()
+        else:
+            return super(NpEncoder, self).default(obj)
+
+
+class DatasetInspector:

```

Comment:
Is there any reason we aren't using a nvtabulat.ops.StatOperator interface here?

Reply:
So you thinking we should use the categorify operator for cardinality? Or are you saying we should create stat ops for the stats we are collecting?

Reply:
The latter.

I'm thinking that if we use a StatOperator class to collect these statistics - not only do we standardize the API, we also gain the ability to collect these statistics as part of a workflow which gives us a bunch of extra functionality for free. (like we can use these statistics for the ml ops work, we can measure dataset statistics on the output of a workflow etc).

Reply:
ahh big picture type stuff... got it.

Reply:
also see  https://github.com/NVIDIA/NVTabular/issues/533

Reply:
This is something I was thinking about, and I was discussing with Ronay because it would work great for #533 as Ben is pointing.

I am gonna create a dataset stats operator. Initially as `datastats.py `, please let me know what it would be a better name.